.eslintrc.js

```diff
@@ -55,192 +50,199 @@ const defaultRules = {
-      'testutils/data.server.testutils.js'
-    ],
-    env: {
-      node: true,
-      jasmine: true,
-      mocha: true
-    },
-    parserOptions: {
-      ecmaVersion: 2018
+  parser: 'babel-eslint',
+  overrides: [
+    /**
+     * Overrides for server side app files
+     */
+    {
+      files: [
+        '*.js',
+        'config/**',
+        'migrations/**',
+        'modules/*/server/*.js',
+        'modules/*/server/**/*.js',
+        'testutils/server/**',
+      ],
+      env: {
+        node: true,
+      },
     },
-    rules: es2018rules,
-    globals: {
-      PhusionPassenger: true,
-    }
-  }, {
-    // overrides for client/react code
-    files: [
-      'config/client/**',
-      'config/env/**',
-      'config/webpack/**',
-      'modules/admin/client/**',
-      'modules/core/client/app/config.js',
-      'modules/**/client/components/**',
-      'modules/**/client/api/**',
-      'modules/**/client/utils/**',
-      'modules/core/client/directives/tr-boards.client.directive.js',
-      'modules/core/client/services/photos.service.js',
-      'modules/references/tests/client/**'
-    ],
-    env: {
-      es6: true,
-      browser: true,
+
+    /**
+     * Overrides for client Angular code
+     */
+    {
+      files: [
+        'modules/*/client/*.module.js',
+        'modules/*/client/config/*.js',
+        'modules/*/client/controllers/*.js',
+        'modules/*/client/directives/*.js',
+        'modules/*/client/filters/*.js',
+        'modules/*/client/services/*.js',
+        'modules/core/client/app/config.js',
+        'modules/core/client/app/init.js',
+        'testutils/client/*.js',
+      ],
+      rules: {
+        'angular/component-limit': 0,
+        'angular/controller-as-route': 1,
+        'angular/controller-as-vm': 1,
+        'angular/controller-as': 1,
+        'angular/deferred': 1,
+        'angular/di-unused': 2,
+        'angular/directive-restrict': 0,
+        'angular/empty-controller': 2,
+        'angular/no-controller': 0,
+        'angular/no-inline-template': 0,
+        'angular/no-run-logic': 0,
+        'angular/no-services': 0,
+        'angular/on-watch': 0,
+        'angular/prefer-component': 0,
+        'angular/no-cookiestore': 2,
+        'angular/no-directive-replace': 0,
+        'angular/no-http-callback': 2,
+        'angular/angularelement': 2,
+        'angular/definedundefined': 2,
+        'angular/document-service': 2,
+        'angular/interval-service': 2,
+        'angular/json-functions': 2,
+        'angular/log': 1,
+        'angular/timeout-service': 2,
+        'angular/typecheck-array': 2,
+        'angular/typecheck-date': 2,
+        'angular/typecheck-function': 2,
+        'angular/typecheck-number': 2,
+        'angular/typecheck-object': 2,
+        'angular/typecheck-string': 2,
+        'angular/window-service': 2
+      },
+      settings: {
+        angular: 1,
+      },
+      env: {
+        browser: true,
```

Comment:
We actually might want to turn this `browser` env off and simply allow `document` and `window` globals.

That will stop nasty bugs like where someone uses `find()` and forgets to import it from `lodash`, causing `window.find` to be referenced and nothing would warn about it — it would just silently break.

Good for another PR.